src/r-exasol/impl/rconnection/RWriterConnection.h

```diff
@@ -0,0 +1,30 @@
+//
+// Created by thomas on 08/07/2021.
+//
+
+#ifndef R_EXASOL_RWRITERCONNECTION_H
+#define R_EXASOL_RWRITERCONNECTION_H
+
+#include <r-exasol/impl/rconnection/RconnFwd.h>
+#include <r-exasol/impl/rconnection/RConnection.h>
+#include <Rdefines.h>
+#include <r-exasol/if/Writer.h>
+
+namespace exa {
+    namespace rconnection {
+        class RWriterConnection : public RConnection {
+        public:
+            explicit RWriterConnection(writer::Writer & writer);
+
+            SEXP create();
+            void release() override;
+
+        private:
+            writer::Writer & mWriter;
+            Rconnection mConn;
```

Comment:
We really need a better distinction between Rconnection (which comes from R I guess and seems something which we can copy) and RConnection which we have defined. Maybe, we could make it more explicit with ExaRConnection or CppRConnection.

Reply:
I started to use namespace for that. Rconnection => ::Rconnection. I think that helps to understand better which type comes from where....